.github/workflows/kind.yaml

```diff
@@ -82,9 +82,9 @@ jobs:
 
       - name: Relay Port Forward
         run: |
-          pkill -f "kubectl port-forward.*hubble-relay" || true
+          pkill -f "cilium hubble port-forward" || true
```

Comment:
I think the `pkill` has failed 🤔 (even though it's not reported as failure due to `|| true`)
```
Error: Unable to port forward: unable to execute "kubectl port-forward -n kube-system svc/hubble-relay --address 0.0.0.0 --address :: 4245:80": exit status 1
❌ Unable to execute "kubectl port-forward -n kube-system svc/hubble-relay --address 0.0.0.0 --address :: 4245:80":
 Unable to listen on port 4245: Listeners failed to create with the following errors: [unable to create listener: Error listen tcp4 0.0.0.0:4245: bind: address already in use unable to create listener: Error listen tcp6 [::]:4245: bind: address already in use]
error: unable to listen on any of the requested ports: [{4245 4245}]
```

Reply:
Actually, why do we even a `|| true`. We should never expect it fail to kill the previous `port-forward`.

Reply:
Fixed in latest version.

Reply:
> Actually, why do we even have a || true. We should never expect it fail to kill the previous port-forward.

Can't it fail to setup port forwarding the first time, in which case we would fail to kill the process?

Reply:
Related: https://github.com/cilium/cilium/issues/16636.

Reply:
I was unaware of this change here - I guess this might conflict with https://github.com/cilium/cilium/pull/16637

Reply:
I think `|| true` (or `|| echo "nothing killed`) is needed, because `pkill` will return a non-zero value of the process already exited by itself due to `cilium uninstall` already killing the pod.

Reply:
I just discovered that `cilium uninstall` will not cause `kubectl port-forward` to terminate. So I think _not_ using `|| true` might actually be correct here, because if there is no running `port-forward` process, then something went wrong.

Reply:
I agree, but we should then surface the failure to run `kubectl port-forward` in the first place.

Reply:
> Can't it fail to setup port forwarding the first time, in which case we would fail to kill the process?

In that case the job will fail at the port-forward step and should not reach the second port-forward step.

> I agree, but we should then surface the failure to run `kubectl port-forward` in the first place.

Can you clarify, I'm not sure what you meant? 🤔

Reply:
FWIW, the `kubectl port-forward` failures I observed in `cilium/cilium` didn't surface because we are using `kubectl port-forward &` to run the command in the background. So there is never a non-zero exit code.
And because we are using `flow-validation=warning`, even the tests using the port-forward didn't fail as any Hubble Relay errors were ignored. The failure only finally surfaced when `pkill`/`pgrep` never found any process.

Reply:
Ah, if that is the issue then we can just retrieve the `&` exit code and expose it as shell return code.

Reply:
I've opened #390 to fix this. It relies on this PR, so I propose we merge it and continue discussion in #390.

Reply:
I'm not sure I understand the need for a separate PR, but I'm fine with that approach if it helps :shrug:

Reply:
I just didn't want to clutter this PR with something unrelated ^^